### ISSUE TITLE
Exception in Accord.Neuro.dll

### ISSUE THREAD

**Issue by HamidAhmadzadegan** [accord-net/framework#119]
Hi
i use Accord.net framework for creating neural network
I create  a new ActivationNetwork and LevenbergMarquardtLearning teacher
and on creating teacher i have this runtime error:
'An unhandled exception of type 'System.OutOfMemoryException' occurred in Accord.Neuro.dll'
my network has 957 nero in input layer and 100 nero in hidden layer and 1 nero in output layer
(ofcourse i need 5000 input nero approximatly, i decrease input nero to solve problem but i have the error yet even with 957 nero in input layer)
my laptop has 4GB memory and when i give this error i have 2GB of free memory (i see in windows task manager)
but accord sample with small data work good
please help me :)
thanks

**Comment by LEVANHAU** [comment]
You can use accord+ afoge for it. May be contac me for this problem

**Comment by LEVANHAU** [comment]
You can try this. I use 7000 for input layer
int rows = dgvTrainingSource.Rows.Count;
             double[][] input = new double[rows][];
              labels = new int[rows];  
             for (int i = 0; i < rows; i++)
            {              
                
                input[i] = (double[])dgvTrainingSource.Rows[i].Cells["colTrainingFeatures"].Value;
                 labels[i] = (int)dgvTrainingSource.Rows[i].Cells["colTrainingLabel"].Value;
          
            }
            //Accord.Statistics. 
             double[][] output = Accord.Statistics.Tools.Expand(labels, 10, -1, +1);
             var function = new BipolarSigmoidFunction(2);
             network = new AForge.Neuro.ActivationNetwork(function , 1024, 10);
             AForge.Neuro.Learning.PerceptronLearning learning = new AForge.Neuro.Learning.PerceptronLearning(network); 

            
            int iteration = 0;
             lbStatus.Text = "Training the classifiers. This may take a (very) significant amount of time...";
            Application.DoEvents();
             
            Stopwatch sw = Stopwatch.StartNew();
            
            double error = Double.PositiveInfinity;
            for (int i = 0; i < 10; i++)
            {
                error = learning.RunEpoch(input, output);   
            }


            sw.Stop();
            lbStatus.Text = String.Format(
                "Training complete ({0}ms, {1}er). Click Classify to test the classifiers.",
                sw.ElapsedMilliseconds, iteration);

**Comment by HamidAhmadzadegan** [comment]
Thank you  LEVANHAU
i need LevenbergMarquardtLearning
i use PerceptronLearning with 11000 input neuro and don't have any problem
but LevenbergMarquardtLearning make a Hessian matrix that is so large and make outOfMemory exception
what about accord+ ? what is it? how can i use of accord+? please tell me more about accord+
Thanks a lot... :)

**Comment by LEVANHAU** [comment]
oh when i use LevenbergMarquardtLearning I had error same you then i chage it to PerceptronLearning

**Comment by cesarsouza** [comment]
Hi Hamid!

In this case, please use ParallelResilientBackpropagationLearning instead. The problem is that Levenberg-Marquardt is mostly meant to be used with small to moderate size problems, as it requires lots of memory. If you need to work with a large number of samples, then its better to stick with first order methods instead, such as resilient backpropagation. 

Best regards,
Cesar
